### PR TITLE
Feature/splunkhec

### PR DESCRIPTION
**Pull request by gyrospectre**
Fixes to the HEC rotation code to make it work with Splunk Cloud as well as Enterprise. Extensive testing with SC to get this working properly.

Added documentation and tests.

### CODE REVIEW

File: src/keydra/clients/splunk.py [hunk 1]
```diff
@@ -245,74 +250,184 @@ def rotate_hectoken(self, inputname):
...
-            raise Exception(
-                'Error rotating HEC token {} on Splunk host '
-                '{}. Input was not found! Input list: {}'.format(
-                    inputname,
-                    self._service.host,
-                    inputlist
+        if tasks[0]['name'] == 'lastDeploy':
+            return tasks[0]['content']['taskId']
+        else:
+            raise Exception('Could not fetch last task Id!')
```
Comment (vludax) [inline src/keydra/clients/splunk.py]: Do we know why? Is it in the response?

File: src/keydra/clients/splunk.py [hunk 2]
```diff
@@ -245,74 +250,184 @@ def rotate_hectoken(self, inputname):
...
+        :returns: New token value if successful
+        :rtype: :class:`string`
+
+        '''
+        LOGGER.debug('Getting input details for {}'.format(inputname))
+
+        inputs = self._get_splunkcloud_httpinput(inputname)
+
+        if len(inputs) == 0:
+            raise Exception('Input was not found!')
```
Comment (vludax) [inline src/keydra/clients/splunk.py]: May want to output the input name, as debug level is usually disabled

**Review by vludax — DISMISSED** [review]

File: src/keydra/clients/splunk.py (hunk 1 above)
Comment (gyrospectre) [reply src/keydra/clients/splunk.py]: This should not happen, and would be a strange scenario where the call was successful but the response did not match the documented format. At any rate, fair point, updated to be slightly more helpful in troubleshooting.

File: src/keydra/clients/splunk.py (hunk 2 above)
Comment (gyrospectre) [reply src/keydra/clients/splunk.py]: Updated!

**Review by vludax — APPROVED** [review]

**Pull request merged**